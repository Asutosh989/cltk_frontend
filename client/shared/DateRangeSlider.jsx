import { debounce } from 'throttle-debounce';
import 'ion-rangeslider/js/ion.rangeSlider.js';
import 'ion-rangeslider/css/ion.rangeSlider.css';
import 'ion-rangeslider/css/ion.rangeSlider.skinNice.css';

DateRangeSlider = React.createClass({

	propTypes: {
		handleChangeDate: React.PropTypes.func.isRequired,
		min: React.PropTypes.number,
		max: React.PropTypes.number,
	},

	componentDidMount() {
		$('#date-range').ionRangeSlider({
			type: 'double',
			min: this.props.min,
			max: this.props.max,
			grid: true,
			postfix: '',
			values_separator: ' to ',
			onChange: debounce(500, this.props.handleChangeDate),

		});
	},

	render() {
		return (
			<div id="date-range" />
		);
	},

});
