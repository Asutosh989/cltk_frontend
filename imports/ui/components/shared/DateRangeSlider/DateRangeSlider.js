import React from 'react';
import PropTypes from 'prop-types';
import { debounce } from 'throttle-debounce';

class DateRangeSlider extends React.Component {

	componentDidMount() {
		$('#date-range').ionRangeSlider({
			type: 'double',
			min: this.props.min,
			max: this.props.max,
			grid: true,
			postfix: '',
			values_separator: ' to ',
			onChange: debounce(500, this.props.handleChangeDate),

		});
	}

	render() {
		return (
			<div id="date-range" />
		);
	}
}

DateRangeSlider.propTypes = {
	handleChangeDate: PropTypes.func.isRequired,
	min: PropTypes.number,
	max: PropTypes.number,
};

export default DateRangeSlider;
