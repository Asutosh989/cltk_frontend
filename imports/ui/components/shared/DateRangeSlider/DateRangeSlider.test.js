
import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// component:
import DateRangeSlider from './DateRangeSlider';

describe('DateRangeSlider', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<MuiThemeProvider>
					<DateRangeSlider />
				</MuiThemeProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
