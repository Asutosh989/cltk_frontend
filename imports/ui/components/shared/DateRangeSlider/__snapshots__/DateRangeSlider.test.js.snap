// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangeSlider renders correctly 1`] = `
<div
  id="date-range"
/>
`;
